Src/DependencyCollector/Shared.Tests/Implementation/EventHubsDiagnosticListenerTests.cs

```diff
@@ -0,0 +1,168 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.DependencyCollector.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+#if !NETCORE
+    using Microsoft.ApplicationInsights.Web.TestFramework;
+#else
+    using Microsoft.ApplicationInsights.Tests;
+#endif
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class EventHubsDiagnosticListenerTests
+    {
+        private TelemetryConfiguration configuration;
+        private List<ITelemetry> sentItems;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            this.configuration = new TelemetryConfiguration();
+            this.sentItems = new List<ITelemetry>();
+            this.configuration.TelemetryChannel = new StubTelemetryChannel { OnSend = item => this.sentItems.Add(item), EndpointAddress = "https://dc.services.visualstudio.com/v2/track" };
+            this.configuration.InstrumentationKey = Guid.NewGuid().ToString();
+        }
+
+        [TestCleanup]
+        public void CleanUp()
+        {
+            while (Activity.Current != null)
+            {
+                Activity.Current.Stop();
+            }
+        }
+
+        [TestMethod]
+        public void EventHubsSuccessfulSendIsHandled()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.IncludeDiagnosticSourceActivities.Add("Microsoft.Azure.EventHubs");
+                module.Initialize(this.configuration);
+
+                DiagnosticListener listener = new DiagnosticListener("Microsoft.Azure.EventHubs");
+
+                Activity parentActivity = new Activity("parent").AddBaggage("k1", "v1").Start();
+                var telemetry  = this.TrackOperation<DependencyTelemetry>(listener, "Microsoft.Azure.EventHubs.Send", TaskStatus.RanToCompletion);
+
+                Assert.IsNotNull(telemetry);
+                Assert.AreEqual("Send", telemetry.Name);
+                Assert.AreEqual(RemoteDependencyConstants.AzureEventHubs, telemetry.Type);
+                Assert.AreEqual("sb://eventhubname.servicebus.windows.net/ | ehname", telemetry.Target);
```

Comment:
Can you describe this in the original GH issue opened? 
https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/753

Specifically - mention which properties goes to which field of DependencyTelemetry. Like target will be ehurl | ehname etc.
(Since our documentations are generally weak, GH issue is only we have to explain to customers about features:) )

Reply:
Done